crypto/evp/evp_enc.c

```diff
@@ -62,13 +62,20 @@ int EVP_CIPHER_CTX_reset(EVP_CIPHER_CTX *ctx)
     ENGINE_finish(ctx->engine);
 #endif
     memset(ctx, 0, sizeof(*ctx));
-    ctx->iv_len = 0;
+    ctx->iv_len = -1;
     return 1;
 }
 
 EVP_CIPHER_CTX *EVP_CIPHER_CTX_new(void)
 {
-    return OPENSSL_zalloc(sizeof(EVP_CIPHER_CTX));
+    EVP_CIPHER_CTX *ctx;
+
+    ctx = OPENSSL_zalloc(sizeof(EVP_CIPHER_CTX));
```

Comment:
Style comment here (which can be ignored if you wish) but when you only have a single line to follow there is no need to have a separate return ... i.e. the following is cleaner looking code IMHO.

```
EVP_CIPHER_CTX *EVP_CIPHER_CTX_new(void)
{
    ctx = OPENSSL_zalloc(sizeof(EVP_CIPHER_CTX));
    if (ctx != NULL)
      ctx->iv_len = -1;

    return ctx;
}

```

Reply:
IMO the current way is more clear.